Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+            var accountResponse = MakeRequest<Account>(requestString);
+            //TODO figure how exchange rates work in Oanda. probably need to call http://developer.oanda.com/exchange-rates-api/v1/currencies/
+            cash.Add(new Cash(accountResponse.accountCurrency, accountResponse.balance.ToDecimal(), new decimal(1.0)));
+            return cash;
+        }
+        
+        private static string GetCommaSeparatedList(IEnumerable<string> items)
+        {
+            var result = new StringBuilder();
+            foreach (var item in items)
+            {
+                result.Append(item + ",");
+            }
+            return result.ToString().Trim(',');
+        }
+
+
+        private Dictionary<string, SecurityType> MapInstrumentToSecurityType(List<Instrument> instruments)
+        {
+            var result = new Dictionary<string, SecurityType>();
+            foreach (var instrument in instruments)
+            {
+                var isForex = Regex.IsMatch(instrument.instrument, "[A-Z]{3}_[A-Z]{3}") && Regex.IsMatch(instrument.displayName, "[A-Z]{3}/[A-Z]{3}");
+                result.Add(instrument.instrument, isForex ? SecurityType.Forex : SecurityType.Cfd);
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the list of available tradable instruments/products from Oanda
+        /// </summary>
+        /// <returns></returns>
+        public List<Instrument> GetInstrumentsAsync(List<string> instrumentNames = null)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Rates) + "instruments?accountId=" + AccountId;
+            if (instrumentNames != null)
+            {
+                var instrumentsParam = GetCommaSeparatedList(instrumentNames);
+                requestString += "&instruments=" + Uri.EscapeDataString(instrumentsParam);
+            }
+            var instrumentResponse = MakeRequest<InstrumentsResponse>(requestString);
+            var instruments = new List<Instrument>();
+            instruments.AddRange(instrumentResponse.instruments);
+            return instruments;
+        }
+
+        /// <summary>
+        /// Places a new order and assigns a new broker ID to the order
+        /// </summary>
+        /// <param name="order">The order to be placed</param>
+        /// <returns>True if the request for a new order has been placed, false otherwise</returns>
+        public override bool PlaceOrder(Order order)
+        {
+            var requestParams = new Dictionary<string, string>
+            {
+                {"instrument", order.Symbol},
+                {"units", Convert.ToInt32(order.AbsoluteQuantity).ToString()}
+            };
+
+            PopulateOrderRequestParameters(order, requestParams);
+
+            Log.Trace(string.Format("OandaBrokerage.PlaceOrder(): {0} to {1} {2} units of {3}", order.Type, order.Direction, order.Quantity, order.Symbol));
```

Comment:
Indeed, I think since then I've added the ToString override, thanks for updating :D